apps/spruce/src/pages/waterfall/VersionLabel.tsx

```diff
@@ -0,0 +1,106 @@
+import { useProjectHealthAnalytics } from "analytics/projectHealth/useProjectHealthAnalytics";
+import { StyledRouterLink } from "components/styles";
+import { getVersionRoute, getTriggerRoute } from "constants/routes";
+import { WaterfallQuery } from "gql/generated/types";
+import { useSpruceConfig, useDateFormat } from "hooks";
+import { shortenGithash } from "utils/string";
+import { jiraLinkify } from "utils/string/jiraLinkify";
+
+type VersionFields = NonNullable<
+  Unpacked<WaterfallQuery["waterfall"]["versions"]>["version"]
+>;
+
+export const VersionLabel: React.FC<VersionFields> = ({
+  author,
+  createTime,
+  gitTags,
+  id,
+  message,
+  revision,
+  upstreamProject,
+}) => {
+  const getDateCopy = useDateFormat();
+  const createDate = new Date(createTime);
+
+  const spruceConfig = useSpruceConfig();
+  const jiraHost = spruceConfig?.jira?.host ?? "";
+
+  const { sendEvent } = useProjectHealthAnalytics({
+    page: "Waterfall",
+  });
+
+  return (
+    <VersionContainer>
+      <Body>
+        <InlineCode
+          as={Link}
+          onClick={() => {
+            sendEvent({
+              name: "Clicked commit label",
+              "commit.type": "active",
+              link: "githash",
+            });
+          }}
+          to={getVersionRoute(id)}
+        >
+          {shortenGithash(revision)}
+        </InlineCode>{" "}
+        {getDateCopy(createDate, { omitSeconds: true, omitTimezone: true })}
+      </Body>
+      {upstreamProject && (
+        <Body>
+          Triggered by:{" "}
+          <StyledRouterLink
+            onClick={() => {
+              sendEvent({
+                name: "Clicked commit label",
+                "commit.type": "active",
+                link: "upstream project",
+              });
+            }}
+            to={getTriggerRoute({
+              triggerType: upstreamProject.triggerType,
+              upstreamTask: upstreamProject.task,
+              upstreamVersion: upstreamProject.version,
+              upstreamRevision: upstreamProject.revision,
+              upstreamOwner: upstreamProject.owner,
+              upstreamRepo: upstreamProject.repo,
+            })}
+          >
+            {upstreamProject.project}
+          </StyledRouterLink>
+        </Body>
+      )}
+      {/* @ts-expect-error */}
+      <CommitMessage title={message}>
+        <strong>{author}</strong> &bull;{" "}
+        {jiraLinkify(message, jiraHost, () => {
+          sendEvent({
+            name: "Clicked commit label",
+            "commit.type": "active",
+            link: "jira",
+          });
+        })}
+      </CommitMessage>
+      {gitTags && <Body>Git Tags: {gitTags.map((g) => g.tag).join(", ")}</Body>}
+    </VersionContainer>
+  );
+};
+
+const VersionContainer = styled.div`
+  > * {
+    font-size: 12px;
+    line-height: 1.3;
+  }
+`;
+
+const CommitMessage = styled(Body)`
+  display: -webkit-box;
+  -webkit-box-orient: vertical;
```

Comment:
[The weird webkit flexbox way](https://css-tricks.com/line-clampin/#aa-the-standardized-way) 😭